prepline_general/api/general.py

```diff
@@ -120,33 +130,55 @@ def partition_file_via_api(file_tuple, **kwargs):
     return elements
 
 
-def partition_pdf_splits(file, **kwargs):
+def partition_pdf_splits(
+    request, file, file_filename, content_type, coordinates, **partition_kwargs
+):
     """
-    Split up a pdf and process in parallel by sending back into the api
+    Split a pdf into chunks and process in parallel with more api calls, or partition
+    locally if the chunk is small enough. As soon as any remote call fails, bubble up
+    the error.
+
+    Arguments:
+    request is used to forward relevant headers to the api calls
+    file, file_filename and content_type are passed on in the file argument to requests.post
+    coordinates is passed on to the api calls, but cannot be used in the local partition case
+    partition_kwargs holds any others parameters that will be forwarded, or passed to partition
     """
     pages_per_pdf = 1
     pdf = PdfReader(file)
 
     # If it's small enough, just process locally
     if len(pdf.pages) <= pages_per_pdf:
-        return partition(file=file, **kwargs)
+        return partition(
+            file=file, file_filename=file_filename, content_type=content_type, **partition_kwargs
+        )
 
     results = []
-    pages = get_pdf_splits(pdf, split_size=pages_per_pdf)
-    partition_func = partial(partition_file_via_api, **kwargs)
+    page_tuples = get_pdf_splits(pdf, split_size=pages_per_pdf)
+
+    partition_func = partial(
+        partition_file_via_api,
+        request=request,
+        filename=file_filename,
+        content_type=content_type,
+        coordinates=coordinates,
+        **partition_kwargs,
+    )
 
     with ThreadPoolExecutor() as executor:
-        for result in executor.map(partition_func, pages):
+        for result in executor.map(partition_func, page_tuples):
             results.extend(result)
 
     return results
 
 
 def pipeline_api(
     file,
+    request=None,
     filename="",
     m_strategy=[],
     m_coordinates=[],
+    m_use_parallel_mode=[],
```

Comment:
Ah, we should not be exposing this to end users. In other words, an end (external) user should not be able to determine the behavior by setting this.

Reply:
Makes sense! I was looking for a good way to run tests against parallel mode. For the new test script, I suppose we can start two containers side by side, but without a runtime parameter we wouldn't be able to compare parallel/non-parallel responses against our hosted api without a manual restart. I'll make a note of that in the test script.

Reply:
yeah, i was assuming we'd just restart the web server to test consistency between the two modes.